test/compress/functions.js

```diff
@@ -3595,6 +3802,45 @@ hoisted_single_use: {
     ]
 }
 
+inlined_single_use: {
+    options = {
+        inline: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        console.log(function(f) {
+            f();
+        }(function() {
+            var a = function() {
+                A;
+            };
+            var b = function() {
+                a(B);
+            };
+            (function() {
+                b;
+            });
+            var c = 42;
+        }));
+    }
```

Comment:
> Which of the test cases was derived from it?

`--validate` failure, so no need to actually `.run_code()` &minus; still pretty cool to reduce such a humongous AST 🧐